src/lib.rs

```diff
@@ -184,7 +185,11 @@ pub fn run_book(
   run_opts: RunOpts,
   warning_opts: WarningOpts,
   compile_opts: CompileOpts,
+  args: Option<Vec<Term>>,
 ) -> Result<(Term, RunInfo), Info> {
+  let mut ctx = Ctx::new(&mut book);
+  ctx.set_entrypoint();
```

Comment:
Why set the entrypoint here and them again inside desugar_book?

Reply:
If I don't set the entrypoint earlier in the run mode it won't be able to apply the cli arguments

Reply:
I see, idk how to fix that is_some() problem then.

maybe the args could be passed to be compiled too? That way set_entrypoint and apply_args would only be called one time inside the compile_book/desugar

Reply:
Idk if accepting args on compile mode would be something we want, but is there a problem with that?

Reply:
If we are compiling we just desugar the patterns, the args are be passed on hvm-core side

Reply:
I think she meant having the compile/desugar function accept arguments, but the cli then just passes None.

run_book -> args = Some(vec![])
compile/desugar_book -> args = None